apiserver/facades/client/controller/controller.go

```diff
@@ -1037,61 +987,54 @@ users to the destination controller or remove them from the current model:
 	return nil
 }
 
-func makeModelInfo(ctx context.Context, st, ctlrSt *state.State,
+func makeModelInfo(ctx context.Context,
+	modelInfo coremodel.Model,
 	controllerConfigService ControllerConfigService,
+	agentService AgentService,
+	accessService ControllerAccessService,
+	apiUser string,
 	modelExporter ModelExporter,
 	store objectstore.ObjectStore,
 	leaders map[string]string,
 ) (coremigration.ModelInfo, userList, error) {
 	var empty coremigration.ModelInfo
 	var ul userList
 
-	model, err := st.Model()
-	if err != nil {
-		return empty, ul, errors.Trace(err)
-	}
-
 	description, err := modelExporter.ExportModel(ctx, leaders, store)
 	if err != nil {
 		return empty, ul, errors.Trace(err)
 	}
 
-	users, err := model.Users()
+	users, err := accessService.GetModelUsers(ctx, apiUser, modelInfo.UUID)
```

Comment:
I am concerned about this call and think that it may need to be reworked. Also seems like this call should perhaps live on the model service. Have asked @Aflynn50 to look at this.

I don't think this should stop this PR at the moment.